bands/site_sections/bands.py

```diff
@@ -9,58 +9,70 @@ def index(self, session, id, message=''):
             'band': session.band(id)
         }
 
-    def agreement(self, session, message='', **params):
-        band = session.band(params, restricted=True)
+    def agreement(self, session, id, message='', **params):
+        band = session.band(id)
```

Comment:
This structure of loading two objects into the session seems a little cumbersome, but it does have the advantage of actually working.